pkg/gatewayserver/scheduling/sub_band.go

```diff
@@ -212,3 +212,23 @@ func (sb *SubBand) ScheduleAnytime(d time.Duration, next func() ConcentratorTime
 	sb.emissions = sb.emissions.Insert(em)
 	return em, nil
 }
+
+// HasOverlap checks if the two sub bands have an overlap.
+func (sb *SubBand) HasOverlap(subBand *SubBand) bool {
+	if sb.MinFrequency < subBand.MinFrequency && subBand.MinFrequency < sb.MaxFrequency {
+		return true
+	} else if sb.MinFrequency < subBand.MaxFrequency && subBand.MaxFrequency < sb.MaxFrequency {
+		return true
+	}
+	if subBand.MinFrequency < sb.MinFrequency && sb.MinFrequency < subBand.MaxFrequency {
+		return true
+	} else if subBand.MinFrequency < sb.MaxFrequency && sb.MaxFrequency < subBand.MaxFrequency {
+		return true
+	}
+	return false
+}
```

Comment:
This one can be a bit simpler I think; https://github.com/TheThingsNetwork/lorawan-stack/blob/master/pkg/gatewayserver/scheduling/emission.go#L82-L87

Reply:
This is super cool!